backend/open_webui/routers/openai.py

```
@@ -762,16 +812,27 @@ async def generate_chat_completion(
+            ):
+                detail = f"Server connection error: {str(e)}"
+            elif isinstance(e, asyncio.TimeoutError):
+                detail = "Request timeout - the server took too long to respond"
+            else:
+                detail = "Open WebUI: Server Connection Error"
 
         raise HTTPException(
             status_code=r.status if r else 500,
-            detail=detail if detail else "Open WebUI: Server Connection Error",
+            detail=detail,
         )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)